### PR TITLE
.NET Standard 2.0, for great justice.

### PR DESCRIPTION
**Pull request by NickCraver**
There are many changes rolled up here around .NET Core 2.
- New `netstandard2.0` build
  - Adding all missing functionality from `netstandard1.x` back in for 2.0 (yay APIs came back!)
- Removed `net40` build (wasn't working anyway)
- TVP return in `netstandard1.3`
- Move from `COREFX` to `NETSTANDARD...` identifiers (COREFX is no longer specific enough) 
- Static/Global test fixes (via xUnit `2.3.0-beta5`)
- Benchmark.net upgrade (and attribute changes for it)
- Adds `NuGet.config`
  - Adds xUnit MyGet feed (`2.3.0-beta5` isn't on NuGet.org yet)
- Fixes a few unit tests with overlapping names (now an xUnit analyzer warning)
- Appveyor fixes to build on latest SDK (image change back to stable).

### CODE REVIEW

File: Dapper.Tests/Dapper.Tests.csproj [hunk 1]
```diff
@@ -6,38 +6,36 @@
     <GenerateDocumentationFile>false</GenerateDocumentationFile>
     <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
     <GenerateBindingRedirectsOutputType>true</GenerateBindingRedirectsOutputType>
-    <TargetFrameworks>net452;netcoreapp1.0</TargetFrameworks>
+    <TargetFrameworks>net452;netcoreapp1.0;netcoreapp2.0</TargetFrameworks>
+    <TargetFrameworks>netcoreapp1.0</TargetFrameworks>
```
Comment (PureKrome) [inline Dapper.Tests/Dapper.Tests.csproj]: Is this 2nd row required?

Comment (NickCraver) [inline Dapper.Tests/Dapper.Tests.csproj]: Nope. Good catch! I was proving out a bug in xUnit beta 4 and forgot to remove - will do so after this set of cleanup commits.

Comment (PureKrome) [inline Dapper.Tests/Dapper.Tests.csproj]: Oh nice! Woot!!

Comment (NickCraver) [inline Dapper.Tests/Dapper.Tests.csproj]: Removed in bdebedea412a5144f962f86a4fcf7e6677870673 that'll immediately follow this PR.

**Comment by NickCraver** [comment]
Merging down, netstandard2-cleanup branch to follow.

**Comment by NickCraver** [comment]
Released on NuGet: https://www.nuget.org/packages/Dapper/1.50.4-alpha1-00070

**Pull request merged**